src/webgpu/shader/execution/expression/case.ts

```diff
@@ -25,6 +27,41 @@ export type Case = {
 /** CaseList is a list of Cases */
 export type CaseList = Array<Case>;
 
+/**
+ * Filters a given set of Cases down to a target number of cases by
+ * randomly selecting which Cases to return.
+ *
+ * The selection algorithm is deterministic and stable for a case's
+ * inputs.
+ *
+ * This means that if a specific case is selected is not affected by the
+ * presence of other cases in the list, so in theory it is possible to create a
+ * pathological set of cases such that all or not of the cases are selected
+ * in spite of the target number.
+ *
+ * This is a trade-off from guaranteeing stability of the selected cases over
+ * small changes, so the target number of cases is more of a suggestion. It is
+ * still guaranteed that if you set n0 < n1, then the invocation with n0 will
+ * return at most the number of cases that n1 does, it just isn't guaranteed to
+ * be less.
+ *
+ * @param n number of cases targeted be returned. Expected to be a positive
+ *          integer. If equal or greater than the number of cases, then all the
+ *          cases are returned. 0 is not allowed, since it is likely a
+ *          programming error, because if the caller intentionally wants 0
+ *          items, they can just use [].
+ * @param cases list of Cases to be selected from.
+ */
+export function selectNCases(n: number, cases: CaseList): CaseList {
+  assert(n > 0 && Math.round(n) === n, `n ${n} is expected to be a positive integer`);
+  const count = cases.length;
+  if (n >= count) {
+    return cases;
+  }
+
+  return cases.filter(c => n * (0xffff_ffff / count) > crc32(c.input.toString()));
```

Comment:
Idea: Should we xor-in another `crc32` of a string parameter (i.e. cache key name)? I'm thinking that we'll get the same N-cases selected for different tests with the same parameters. XOR-ing in something unique to the test would prevent this.

Reply:
That is a good point